site/gatsby-site/migrations/2024.12.11T14.39.38.refactor-reports-translations.js

```diff
@@ -0,0 +1,77 @@
+/** @type {import('umzug').MigrationFn<any>} */
+exports.up = async ({ context: { client } }) => {
+  const db = client.db('translations');
+
+  const reportsEnCollection = db.collection('reports_en');
+
+  const reportsEsCollection = db.collection('reports_es');
+
+  const reportsFrCollection = db.collection('reports_fr');
+
+  const reportsJaCollection = db.collection('reports_ja');
+
+  const reportsCollection = db.collection('reports');
+
+  // Fetch all documents from reports_en, reports_es, reports_fr, and reports_ja
+  const reportsEn = await reportsEnCollection.find().toArray();
+
+  const reportsEs = await reportsEsCollection.find().toArray();
+
+  const reportsFr = await reportsFrCollection.find().toArray();
+
+  const reportsJa = await reportsJaCollection.find().toArray();
+
+  // Transform and insert documents into reports with a new field "language"
+  const transformedReportsEn = reportsEn.map((report) => ({
+    ...report,
+    language: 'en',
+  }));
+
+  const transformedReportsEs = reportsEs.map((report) => ({
+    ...report,
+    language: 'es',
+  }));
+
+  const transformedReportsFr = reportsFr.map((report) => ({
+    ...report,
+    language: 'fr',
+  }));
+
+  const transformedReportsJa = reportsJa.map((report) => ({
+    ...report,
+    language: 'ja',
+  }));
+
+  // Insert transformed documents into reports
+  const resultEn = await reportsCollection.insertMany(transformedReportsEn);
```

Comment:
I wonder if this will be too big to insert at once. Have you tried using the production data already?

Reply:
Yes, I executed this migration on my personal MongoDB instance hosted in Atlas, which is a restore of the Production database, and it works fine 👌.